engine/runtime/src/main/java/org/enso/interpreter/runtime/data/atom/GetFieldNode.java

```diff
@@ -1,15 +1,16 @@
 package org.enso.interpreter.runtime.data.atom;
 
-import com.oracle.truffle.api.TruffleLanguage;
 import com.oracle.truffle.api.frame.VirtualFrame;
 import com.oracle.truffle.api.nodes.NodeInfo;
-import com.oracle.truffle.api.nodes.RootNode;
+import org.enso.compiler.context.LocalScope;
 import org.enso.interpreter.EnsoLanguage;
+import org.enso.interpreter.node.EnsoRootNode;
 import org.enso.interpreter.runtime.callable.function.Function;
 import org.enso.interpreter.runtime.data.Type;
+import org.enso.interpreter.runtime.scope.ModuleScope;
 
 @NodeInfo(shortName = "get_field", description = "A base for auto-generated Atom getters.")
-final class GetFieldNode extends RootNode {
+final class GetFieldNode extends EnsoRootNode {
```

Comment:
Changes in `GetFieldNode` may signal that synthetic getters are also checked for `private` access. Are they?

Reply:
I believe so. In pure Enso code, all the runtime private checks are done in `InvokeFunctionNode`, as can be seen for example here: https://github.com/enso-org/enso/blob/f7390133b5df87736aab2a6fc1777e8d5257f710/engine/runtime/src/main/java/org/enso/interpreter/node/callable/dispatch/InvokeFunctionNode.java#L167-L169

But this change was mostly necessary to fix the polyglot access. e.g., when Javascript calls a private constructor via a callback, or when Javascript tries to access the private field. Without this change, the accesses from Javascript would fail on a weird "Truffle context sharing"-related error.